py/persistentcode.c

```diff
@@ -483,6 +487,26 @@ void mp_raw_code_load(mp_reader_t *reader, mp_compiled_module_t *cm) {
         }
     }
 
+    size_t arch_flags = 0;
+    if (MPY_FEATURE_ARCH_FLAGS_TEST(header[2])) {
+        // TODO: Report an error for targets that do not have support for
```

Comment:
I suggest that if the flags bit is set, it should always do the `read_uint`.  That makes the parser more robust.

Then, for non-RV32 archs, those flags should be 0.

Reply:
Originally the read would have been done in all cases, but having an unexpected flag bit is a signal in itself that something's not right with the file and thus it's not right to proceed.

I moved that behind a platform ifdef to reduce the footprint increase, since this affects all ports.  You can check the history of https://github.com/micropython/micropython/pull/18106#issuecomment-3315085609 to see the impact on non-RV32 ports.